resources/aws/recordset.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/aws-operator/resources"
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+type RecordSet struct {
+	Client       *route53.Route53
+	Resource     resources.DNSNamedResource
+	Domain       string
+	HostedZoneID string
+}
+
+// CreateIfNotExists is not implemented because AWS provides UPSERT functionality for DNS records
+func (record RecordSet) CreateIfNotExists() (bool, error) {
+	return false, microerror.MaskAny(notImplementedMethodError)
+}
+
+func (record RecordSet) CreateOrFail() error {
+	if err := record.perform(route53.ChangeActionUpsert); err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	return nil
+}
+
+func (record RecordSet) Delete() error {
+	if err := record.perform(route53.ChangeActionDelete); err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	return nil
+}
+
+func (record RecordSet) perform(action string) error {
+	if record.Client == nil {
+		return clientNotInitializedError
+	}
+
+	params := record.buildParams(action)
+
+	_, err := record.Client.ChangeResourceRecordSets(params)
+
```

Comment:
And bonus points from me for doing

```
if _, err := record.Client.ChangeResourceRecordSets(params); err != nil {
    return microerror.MaskAny(err)
}
```

Not a big deal, but looks cleaner IMO.